src/prefect/orion/schemas/states.py

```diff
@@ -52,7 +52,19 @@ class StateDetails(PrefectBaseModel):
     refresh_cache: bool = None
 
 
-class State(IDBaseModel, Generic[R]):
+class StateBaseModel(IDBaseModel):
+    def orm_dict(
+        self, *args, shallow: bool = False, json_compatible: bool = False, **kwargs
+    ) -> dict:
+        schema_dict = self.dict(
+            *args, shallow=shallow, json_compatible=json_compatible, **kwargs
+        )
+        # remove the data field in order to construct a state ORM model
+        schema_dict.pop("data", None)
```

Comment:
Hm so where does the data go?

Reply:
it disappears, which is why this odd behavior is scoped to just this method; we're only calling `orm_dict` in test fixtures and the like, and I added this method so we aren't automatically dropping `data` whenever we're constructing states in the application